api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -64,74 +55,50 @@ public boolean deleteInstitution(final String shortName) {
-                        addresses.stream()
-                            .map(address -> new DbInstitutionEmailAddress(dbClass, address))
-                            .collect(Collectors.toSet()))));
-
-    return institutionDao.save(dbClass);
+  private DbInstitution updateDbObject(
+      final DbInstitution dbObject, final Institution modelObject) {
+    return dbObject
+        .shortName(modelObject.getShortName())
+        .displayName(modelObject.getDisplayName())
+        .organizationTypeEnum(modelObject.getOrganizationTypeEnum())
+        .organizationTypeOtherText(modelObject.getOrganizationTypeOtherText())
+        .emailDomains(
+            Optional.ofNullable(modelObject.getEmailDomains()).orElse(Collections.emptyList())
+                .stream()
+                .map(domain -> new DbInstitutionEmailDomain().setEmailDomain(domain))
+                .collect(Collectors.toSet()))
+        .emailAddresses(
+            Optional.ofNullable(modelObject.getEmailAddresses()).orElse(Collections.emptyList())
+                .stream()
+                .map(address -> new DbInstitutionEmailAddress().setEmailAddress(address))
+                .collect(Collectors.toSet()));
   }
 
-  private Institution toModelClass(final DbInstitution dbClass) {
-    final Institution institution =
-        new Institution()
-            .shortName(dbClass.getShortName())
-            .displayName(dbClass.getDisplayName())
-            .organizationTypeEnum(
-                DbStorageEnums.organizationTypeFromStorage(dbClass.getOrganizationTypeEnum()))
-            .organizationTypeOtherText(dbClass.getOrganizationTypeOtherText());
-
-    Optional.ofNullable(dbClass.getEmailDomains())
-        .ifPresent(
-            domains ->
-                institution.emailDomains(
-                    domains.stream()
-                        .map(DbInstitutionEmailDomain::getEmailDomain)
-                        .collect(Collectors.toList())));
-
-    Optional.ofNullable(dbClass.getEmailAddresses())
-        .ifPresent(
-            addresses ->
-                institution.emailAddresses(
-                    addresses.stream()
-                        .map(DbInstitutionEmailAddress::getEmailAddress)
-                        .collect(Collectors.toList())));
-
-    return institution;
+  private Institution toModel(final DbInstitution dbObject) {
+    return new Institution()
+        .shortName(dbObject.getShortName())
+        .displayName(dbObject.getDisplayName())
+        .organizationTypeEnum(dbObject.getOrganizationTypeEnum())
+        .organizationTypeOtherText(dbObject.getOrganizationTypeOtherText())
+        .emailDomains(
+            Optional.ofNullable(dbObject.getEmailDomains()).orElse(Collections.emptySet()).stream()
+                .map(DbInstitutionEmailDomain::getEmailDomain)
+                .collect(Collectors.toList()))
+        .emailAddresses(
+            Optional.ofNullable(dbObject.getEmailAddresses()).orElse(Collections.emptySet())
```

Comment:
With the db objects, we can (and I think should) avoid returning null for collections. We don't have a choice with the generated classes though.

Reply:
Actually dbObject.getEmailAddresses() should never be null, since emailAddresses is initialized -> https://github.com/all-of-us/workbench/blob/joel/hibernate-institution/api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java#L28-L28 Hibernate also calls the setEmailAddresses(final Set<DbInstitutionEmailAddress> emailAddresses) with an empty collection if no emailAddresses exist. This test verifies this: https://github.com/all-of-us/workbench/blob/joel/hibernate-institution/api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java#L64-L64

Reply:
Good point - updated